pkg/prometheus/statefulset.go

```diff
@@ -366,14 +369,29 @@ func BuildCommonVolumes(p monitoringv1.PrometheusInterface, tlsAssetSecrets []st
 	return volumes, promVolumeMounts, nil
 }
 
-func VolumeClaimName(p monitoringv1.PrometheusInterface, cpf monitoringv1.CommonPrometheusFields) string {
-	volName := VolumeName(p)
+func VolumeClaimName(p monitoringv1.PrometheusInterface, cpf monitoringv1.CommonPrometheusFields) []string {
+	volNames := []string{}
 	if cpf.Storage != nil {
-		if cpf.Storage.VolumeClaimTemplate.Name != "" {
+		volName := VolumeName(p)
+		if cpf.Storage.VolumeClaimTemplates != nil {
+			for _, pvcTemplate := range cpf.Storage.VolumeClaimTemplates {
+				volName = VolumeName(p)
+				if pvcTemplate.Name != "" {
+					volName = pvcTemplate.Name
+				}
+				volNames = append(volNames, volName)
+			}
+		} else if cpf.Storage.VolumeClaimTemplate.Name != "" {
```

Comment:
The only way we get here is if `cpf.Storage.VolumeClaimTemplates` is nil, right? If that's the case, trying to access `cpf.Storage.VolumeClaimTemplate.Name` will throw a nil pointer exception. 

We should do nothing if VolumeClaimTemplates is nil

Reply:
So how about when user only define `Storage.VolumeClaimTemplate`?

Reply:
Ah sorry, I misread the condition. I thought this line was comparing VolumeClaimTemplate<ins>s</ins>.Name 😅 

You're correct